test/UID2.Client.Test/SharingClientTests.cs

```diff
@@ -0,0 +1,420 @@
+            var json = KeySharingResponse(new[] { MASTER_KEY, SITE_KEY }, IdentityScope.UID2, callerSiteId: 4874, defaultKeysetId: 12345);
+
+            var refreshResult = receivingClient.RefreshJson(json);
+            Assert.True(refreshResult.Success);
+
+            var res = receivingClient.DecryptTokenIntoRawUid(advertisingToken);
+            Assert.True(res.Success);
+            Assert.Equal(EXAMPLE_EMAIL_RAW_UID2_V2, res.Uid);
+        }
+
+        [Fact]
+        public void SharingTokenIsV4()
+        {
+            var client = SharingSetupAndEncrypt(out string advertisingToken);
+
+            char[] base64SpecialChars = { '+', '/', '=' };
+            Boolean containsBase64SpecialChars = advertisingToken.IndexOfAny(base64SpecialChars) != -1;
+            Assert.False(containsBase64SpecialChars);
+        }
+
+        [Fact]
+        public void Uid2ClientProducesUid2Token()
+        {
+            var client = SharingSetupAndEncrypt(out string advertisingToken);
+
+            Assert.Equal("A", advertisingToken.Substring(0, 1));
+        }
+
+
+        [Fact]
+        public void EuidClientProducesEuidToken()
+        {
+            var client = new SharingClient("endpoint", "authkey", CLIENT_SECRET);
+            var json = KeySetToJsonForSharing(new [] {MASTER_KEY, SITE_KEY}, IdentityScope.EUID);
+            var refreshResult = client.RefreshJson(json);
+            Assert.True(refreshResult.Success);
+
+            string advertisingToken = SharingEncrypt(client);
+
+            Assert.Equal("E", advertisingToken.Substring(0, 1));
+        }
+
+        [Fact]
+        public void RawUidProducesCorrectIdentityTypeInToken()
+        {
+            var json = KeySetToJsonForSharing(new [] {MASTER_KEY, SITE_KEY});
+            var refreshResult = _client.RefreshJson(json);
+            Assert.True(refreshResult.Success);
+
+            //see UID2-79+Token+and+ID+format+v3 . Also note EUID does not support v2 or phone
+            Assert.Equal(IdentityType.Email,
+                GetTokenIdentityType("Q4bGug8t1xjsutKLCNjnb5fTlXSvIQukmahYDJeLBtk=",
+                    _client)); //v2 +12345678901. Although this was generated from a phone number, it's a v2 raw UID which doesn't encode this information, so token assumes email by default.
+            Assert.Equal(IdentityType.Phone, GetTokenIdentityType("BEOGxroPLdcY7LrSiwjY52+X05V0ryELpJmoWAyXiwbZ", _client)); //v3 +12345678901
+            Assert.Equal(IdentityType.Email, GetTokenIdentityType("oKg0ZY9ieD/CGMEjAA0kcq+8aUbLMBG0MgCT3kWUnJs=", _client)); //v2 test@example.com
+            Assert.Equal(IdentityType.Email, GetTokenIdentityType("AKCoNGWPYng/whjBIwANJHKvvGlGyzARtDIAk95FlJyb", _client)); //v3 test@example.com
+            Assert.Equal(IdentityType.Email, GetTokenIdentityType("EKCoNGWPYng/whjBIwANJHKvvGlGyzARtDIAk95FlJyb", _client)); //v3 EUID test@example.com
+        }
+
+        private IdentityType GetTokenIdentityType(string rawUid, SharingClient client)
```

Comment:
`client` is unused, what's the purpose of this param?

Reply:
that was taken from EncryptionTestsV4, will fix both